packages/backend-data/src/convert_js_resolvers.ts

```diff
@@ -71,12 +47,38 @@ export const convertJsResolverDefinition = (
 
     const resolverName = `Resolver_${resolver.typeName}_${resolver.fieldName}`;
 
+    const amplifyEnvironmentName =
+      scope.node.tryGetContext('amplifyEnvironmentName') ?? 'NONE';
     new CfnResolver(scope, resolverName, {
       apiId: amplifyApi.apiId,
       fieldName: resolver.fieldName,
       typeName: resolver.typeName,
       kind: APPSYNC_PIPELINE_RESOLVER,
-      codeS3Location: jsResolverTemplateAsset.s3ObjectUrl,
+      /**
+       * The top-level passthrough resolver request/response handler (see: https://docs.aws.amazon.com/appsync/latest/devguide/resolver-reference-overview-js.html#anatomy-of-a-pipeline-resolver-js)
+       * It's required for defining a pipeline resolver. Adds the GraphQL API ID and Amplify environment name to the context stash.
+       * Returns the output of the last function in the pipeline back to the client.
+       *
+       * Customer-provided handlers are added as a Functions list in `pipelineConfig.functions`
+       *
+       * Use inline code to remove circular dependency when adding the API ID.
+       */
+      code: `
+        /**
+         * Pipeline resolver request handler
+         */
+        export const request = (ctx) => {
+          ctx.stash.apiId = '${amplifyApi.apiId}';
```

Comment:
I'd normally recommend we namespace all amplify variables, as in:
- `amplifyAppsyncApiId`
- `amplifyEnvironmentName`
- `amplifyFooBar...`

However! Since we already have an output name for API ID, I wonder if we should use that:

- `awsAppsyncApiId`

For parallelism, that makes the other variable something like:

- `awsAmplifyEnvironmentName`